storage/aptosdb/src/state_store/mod.rs

```diff
@@ -54,7 +54,12 @@ impl StateStore {
         state_key: &StateKey,
         version: Version,
     ) -> Result<(Option<StateValue>, SparseMerkleProof<StateValue>)> {
-        JellyfishMerkleTree::new(self).get_with_proof(state_key.hash(), version)
+        let (state_key_value_option, proof) =
+            JellyfishMerkleTree::new(self).get_with_proof(state_key.hash(), version)?;
+        Ok((
+            state_key_value_option.map(|x| x.value),
+            SparseMerkleProof::new(proof.leaf(), proof.siblings().to_vec()),
```

Comment:
nit: this is unnecessary mem copying. Make a `SparseMerkleProof<V>::into<Self<U>>()`?